frontend/src/bundles/common/store/actions.ts

```diff
@@ -0,0 +1,32 @@
+import { createAsyncThunk } from '@reduxjs/toolkit';
+
+import { storage } from '~/framework/storage/storage.js';
+
+import { applyThemeClassname } from '../helpers/helpers.js';
+import { type Theme, type ValueOf } from '../types/types.js';
+
+const fetchTheme = createAsyncThunk<ValueOf<typeof Theme>, undefined>(
+    'theme/fetchTheme',
+    async () => {
+        const currentTheme = await storage.get('theme');
+        const theme =
+            currentTheme === null
+                ? 'dark'
+                : (currentTheme as ValueOf<typeof Theme>);
+        await storage.set('theme', theme);
+        applyThemeClassname(theme);
+        return theme;
+    },
+);
+
+const setTheme = createAsyncThunk<ValueOf<typeof Theme>, ValueOf<typeof Theme>>(
+    'theme/setTheme',
+    async (theme: ValueOf<typeof Theme>) => {
+        applyThemeClassname(theme);
+        await storage.set('theme', theme);
+
+        return theme;
+    },
+);
```

Comment:
let's do this as toggle theme, not as setTheme

Reply:
Updated the name to toggle theme   https://github.com/BinaryStudioAcademy/bsa-winter-2023-2024-lime/pull/95/commits/8df36008352710c23ce3dbfd8b6f846230b6afa2